src/meta/src/stream/stream_manager.rs

```diff
@@ -483,13 +483,19 @@ impl GlobalStreamManager {
 
         let dummy_table_id = table_fragments.table_id();
 
+        let init_split_assignment = self
+            .source_manager
+            .pre_allocate_splits(&dummy_table_id)
+            .await?;
+
         if let Err(err) = self
             .barrier_scheduler
             .run_config_change_command_with_pause(Command::ReplaceTable {
                 old_table_fragments,
                 new_table_fragments: table_fragments,
                 merge_updates,
                 dispatchers,
+                init_split_assignment,
```

Comment:
have you change source exec to handle this mutation?

Reply:
I add related logic in `command.rs` so `ReplaceTable` will be converted to a `Update` mutation with split info that can be handled by source exec without modifying it.